test/webaudio/audioworklet_post_function.c

```diff
@@ -6,28 +6,121 @@
+  assert(i == 2);
+  assert(j == 3);
+  ++success;
+}
+
+void viii(int i, int j, int k) {
+  assert(testingAudioWorklet == emscripten_current_thread_is_audio_worklet());
+  emscripten_out("viii");
+  assert(i == 4);
+  assert(j == 5);
+  assert(k == 6);
+  ++success;
+}
+
+void vd(double i) {
+  assert(testingAudioWorklet == emscripten_current_thread_is_audio_worklet());
+  emscripten_out("vd");
+  assert(i == 1.5);
+  ++success;
 }
 
-// This event will fire on the audio worklet thread.
-void MessageReceivedInAudioWorkletThread(int a, int b) {
-  printf("MessageReceivedInAudioWorkletThread: a=%d, b=%d\n", a, b);
-  assert(emscripten_current_thread_is_audio_worklet());
-  assert(a == 42 && b == 9000);
-  emscripten_audio_worklet_post_function_viii(EMSCRIPTEN_AUDIO_MAIN_THREAD, MessageReceivedOnMainThread, /*d=*/1, /*e=*/2, /*f=*/3);
+void vdd(double i, double j) {
+  assert(testingAudioWorklet == emscripten_current_thread_is_audio_worklet());
+  emscripten_out("vdd");
+  assert(i == 2.5);
+  assert(j == 3.5);
+  ++success;
+}
+
+void vddd(double i, double j, double k) {
+  assert(testingAudioWorklet == emscripten_current_thread_is_audio_worklet());
+  emscripten_out("vddd");
+  assert(i == 4.5);
+  assert(j == 5.5);
+  assert(k == 6.5);
+  ++success;
+}
+
+void viiiiiidddddd(int a, int b, int c, int d, int e, int f, double g, double h, double i, double j, double k, double l) {
+  assert(testingAudioWorklet == emscripten_current_thread_is_audio_worklet());
+  emscripten_out("viiiiiidddddd");
+  assert(a == 10);
+  assert(b == 11);
+  assert(c == 12);
+  assert(d == 13);
+  assert(e == 14);
+  assert(f == 15);
+  assert(g == 16.5);
+  assert(h == 17.5);
+  assert(i == 18.5);
+  assert(j == 19.5);
+  assert(k == 20.5);
+  assert(l == 21.5);
+  ++success;
+}
+
+void sided_test_finished() {
+  assert(testingAudioWorklet == emscripten_current_thread_is_audio_worklet());
+
+  if (testingAudioWorklet) { // for the first call, finishing checks for main -> audio worklet
+    testingAudioWorklet = false;
+
+    // These event will fire callbacks on main thread.
+    emscripten_audio_worklet_post_function_v(EMSCRIPTEN_AUDIO_MAIN_THREAD, v);
+    emscripten_audio_worklet_post_function_vi(EMSCRIPTEN_AUDIO_MAIN_THREAD, vi, 1);
+    emscripten_audio_worklet_post_function_vii(EMSCRIPTEN_AUDIO_MAIN_THREAD, vii, 2, 3);
+    emscripten_audio_worklet_post_function_viii(EMSCRIPTEN_AUDIO_MAIN_THREAD, viii, 4, 5, 6);
+    emscripten_audio_worklet_post_function_vd(EMSCRIPTEN_AUDIO_MAIN_THREAD, vd, 1.5);
+    emscripten_audio_worklet_post_function_vdd(EMSCRIPTEN_AUDIO_MAIN_THREAD, vdd, 2.5, 3.5);
+    emscripten_audio_worklet_post_function_vddd(EMSCRIPTEN_AUDIO_MAIN_THREAD, vddd, 4.5, 5.5, 6.5);
+    emscripten_audio_worklet_post_function_sig(EMSCRIPTEN_AUDIO_MAIN_THREAD, viiiiiidddddd, "iiiiiidddddd", 10, 11, 12, 13, 14, 15, 16.5, 17.5, 18.5, 19.5, 20.5, 21.5);
+    emscripten_audio_worklet_post_function_v(EMSCRIPTEN_AUDIO_MAIN_THREAD, sided_test_finished);
```

Comment:
Why not use a different function here?  e.g. `do_exit`?    Then you don't need the `testingAudioWorklet` global maybe?

Reply:
1. On resolving conflict with merged PR #23695, I used `emscripten_force_exit(0);` as the PR used, instead of `do_exit`.
2. Originally, `testingAudioWorklet` is on the purpose to check `emscripten_current_thread_is_audio_worklet()` behaves correctly on each callbacks. I changed the name of it to `onAudioWorkletThread`, for legibility.

Reply:
Sure, it just seems a little odd that this callback has two completely different behaviors depending on the thread.   Why not have two different callbacks, one which gets called on the worklet and triggers the second set of callbacks, and one which run on the main thread call `on_exit` which just does the exiting?

Reply:
I agree